trackpy/motion.py

```diff
@@ -184,7 +184,10 @@ def subtract_drift(traj, drift=None):
 
     if drift is None:
         drift = compute_drift(traj)
-    return traj.set_index('frame', drop=False).sub(drift, fill_value=0)
+    traj.set_index('frame', inplace=True, drop=False)
+    for col in drift.columns:
```

Comment:
I should actually time the performance: it's only a loop on 2 or 3 columns. Before, every column got changed to `float` suggesting that all the data was rewritten. Possibly my change made it even faster.

Else: `df.loc[pos_columns].sub(..)` might do the job faster.